lib/badge_queries.rb

```diff
@@ -272,4 +272,31 @@ def self.consecutive_visits(days)
     SQL
   end
 
+  def self.anniversaries(start_date, end_date)
+    start_date = start_date.iso8601(6)
+    end_date = end_date.iso8601(6)
+
+    <<~SQL
+      SELECT u.id
+        FROM users AS u
+        JOIN posts AS p ON p.user_id = u.id
+        JOIN topics AS t ON p.topic_id = t.id
+       WHERE u.id > 0
+         AND u.active
+         AND NOT u.staged
+         AND (u.silenced_till IS NULL OR u.silenced_till < '#{start_date}')
+         AND (u.suspended_till IS NULL OR u.suspended_till < '#{start_date}')
```

Comment:
These two new lines are new, right? Is this a bug fix?

Reply:
We previously had `u.silenced_till IS NULL` but I made it consistent with pretty much every other places where we check "active" users.

These two new lines do change the query a tiny bit though. 

**Before** 

- You could not get the badge if you were currently silenced (no matter what period is being checked)
- You could get the badge if you were suspended 😬 

**After**

- You can't get the badge if you were silenced during the past year
- You can't get the badge if you were suspended during the past year